src/frontend/Felles/Visningskomponenter/UlagretDataModal.tsx

```diff
@@ -1,42 +1,37 @@
 import React, { FC } from 'react';
-import { useNavigate } from 'react-router-dom';
 import { useApp } from '../../App/context/AppContext';
 import { ModalWrapper } from '../Modal/ModalWrapper';
+import ReactRouterPrompt from 'react-router-prompt';
 
 const UlagretDataModal: FC = () => {
-    const {
-        nullstillIkkePersisterteKomponenter,
-        visUlagretDataModal,
-        valgtSide,
-        settVisUlagretDataModal,
-    } = useApp();
-    const navigate = useNavigate();
+    const { nullstillIkkePersisterteKomponenter, ulagretData } = useApp();
 
     return (
-        <ModalWrapper
-            tittel={
-                'Du har ikke lagret dine siste endringer og vil miste disse om du forlater siden'
-            }
-            visModal={visUlagretDataModal}
-            onClose={() => settVisUlagretDataModal(false)}
-            aksjonsknapper={{
-                hovedKnapp: {
-                    onClick: () => settVisUlagretDataModal(false),
-                    tekst: 'Gå tilbake for å lagre',
-                },
-                lukkKnapp: {
-                    onClick: () => {
-                        if (valgtSide) {
-                            nullstillIkkePersisterteKomponenter();
-                            navigate(valgtSide);
-                        }
-                        settVisUlagretDataModal(false);
-                    },
-                    tekst: 'Forlat siden',
-                },
-                marginTop: 4,
-            }}
-        />
+        <ReactRouterPrompt when={ulagretData}>
```

Comment:
Her er "the shit". 

Vi kan godt vurdere å skrive vår egen reactRouterPrompt - er litt spent på setTimeout-issue som _kan_ oppstå... 😬  Skulle gjerne ha gjort nullstill samtidig som confirm, men de kan ikke gå for tidlig/sent. Da blir det 💥

Reply:
Det er ju relativt lite kode 😬 og kun avhengighet til react og router som er fint
https://github.com/sshyam-gupta/react-router-prompt/tree/main/src

Reply:
Ja, men spent på om de fikser oppgraderinger når dette brekker i neste versjon i react-router... 🤔 😬